src/main/java/gregtech/api/items/IDyeableItem.java

```diff
@@ -0,0 +1,88 @@
+package gregtech.api.items;
+
+import net.minecraft.block.BlockCauldron;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+
+import org.jetbrains.annotations.NotNull;
+
+public interface IDyeableItem {
+
+    default boolean hasColor(ItemStack stack) {
+        NBTTagCompound nbttagcompound = stack.getTagCompound();
+        return nbttagcompound != null && nbttagcompound.hasKey("display", Constants.NBT.TAG_COMPOUND) &&
+                nbttagcompound.getCompoundTag("display").hasKey("color", Constants.NBT.TAG_INT);
+    }
+
+    default int getColor(ItemStack stack) {
+        NBTTagCompound nbttagcompound = stack.getTagCompound();
+        if (nbttagcompound != null) {
+            NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("display");
+            if (nbttagcompound1.hasKey("color", Constants.NBT.TAG_INT)) {
+                return nbttagcompound1.getInteger("color");
+            }
+        }
+        return getDefaultColor(stack);
+    }
+
+    default int getDefaultColor(ItemStack stack) {
+        return 0xFFFFFF;
+    }
+
+    default void removeColor(ItemStack stack) {
+        NBTTagCompound nbttagcompound = stack.getTagCompound();
+        if (nbttagcompound != null) {
+            NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("display");
+            if (nbttagcompound1.hasKey("color")) {
+                nbttagcompound1.removeTag("color");
+            }
+        }
+    }
+
+    default void setColor(ItemStack stack, int color) {
+        NBTTagCompound nbttagcompound = stack.getTagCompound();
+        if (nbttagcompound == null) {
+            nbttagcompound = new NBTTagCompound();
+            stack.setTagCompound(nbttagcompound);
+        }
+        NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("display");
```

Comment:
Why have the whole `display` NBT tag, instead of just using an int tag for color? The `display` tag doesn't seem to really be used at all, and seems mostly redundant.